xivo_bus/mixins.py

```diff
@@ -286,22 +336,24 @@ def __init__(self, **kwargs):
             pass
 
     def __run(self, ready_flag, **kwargs):
-        publish = None
         ready_flag.set()
-        with Connection(self.url) as connection:
+        publisher_args = self.queue_publisher_args
+
+        with Connection(self.url, transport_options=publisher_args) as connection:
             while not self.is_stopping or self.__flushing:
-                if not publish:
-                    publish = self.Producer(connection)
                 try:
                     payload, headers, routing_key = self.__fifo.get()
                 except (Empty, TypeError):
                     self.__flushing = False
                     continue
                 try:
-                    publish(payload, headers=headers, routing_key=routing_key)
+                    with self.Producer(connection, **publisher_args) as publish:
+                        publish(payload, headers=headers, routing_key=routing_key)
+                except OperationalError as exc:
+                    self.log.error('Publishing queue error: %s', exc, exc_info=1)
+                else:
                     self.__fifo.task_done()
-                except Exception:
-                    self.log.exception('Error while publishing')
+            connection.release()
```

Comment:
No need to release connection, the `Connection` contextmanager will already release it